File: 824-number-of-lines-to-write-string/solution.cpp
class Solution {
public:
    vector<int> numberOfLines(vector<int>& widths, string S) {
        int countLine = 1, width = 0;
        for (auto & s : S) {
            if (width + widths[s-'a'] <= 100) width += widths[s-'a'];
            else {
                countLine += 1;
                width = widths[s-'a'];
            }
        }
        vector<int> result = {countLine, width};
        return result;
    }
};

File: 824-number-of-lines-to-write-string/solution.py
class Solution(object):
    def numberOfLines(self, widths, S):
        """
        :type widths: List[int]
        :type S: str
        :rtype: List[int]
        """
        count_line, count = 1, 0
        for s in S:
            w = widths[ord(s)-ord('a')] 
            if count + w <= 100:
                count += w
            else:
                count_line += 1
                count = widths[ord(s)-ord('a')]
        return [count_line, count]
